geoAssembler/io_utils.py

```diff
@@ -11,30 +10,33 @@ def read_geometry(detector, filename, quad_pos=None):
     Parameters:
         detector (str): Name of the considered detector
         filename (str): Path to the geometry file
-    Keywords:
         quad_pos (list): X,Y coordinates of quadrants (default None)
 
     Retruns:
         GeometryAssembler Object
     """
-    filename = filename or ''
-    try:
-        quad_pos = quad_pos or Defaults.fallback_quad_pos[detector]
-    except KeyError:
-        raise NotImplementedError('Detector Class not available')
+    if quad_pos is None:
+        quad_pos = Defaults.fallback_quad_pos.get(detector)
 
     if detector == 'AGIPD':
-        from .geometry import AGIPDGeometry
-        if os.path.isfile(filename):
+        if filename:
             return AGIPDGeometry.from_crystfel_geom(filename)
         else:
             return AGIPDGeometry.from_quad_positions(quad_pos)
     elif detector == 'LPD':
```

Comment:
being very nit-picker, DSSC is everywhere define before LPD. In any case, it does not affect the code ;)

Reply:
Thanks, that's attention to detail! :laughing: